### PR TITLE
docs(book): explain concepts required for building a RIOT-rs application

### PR DESCRIPTION
**Pull request by ROMemories**
# Description

A first shot at documenting the usage of Embassy peripherals and RIOT-rs-specific macros.

## Change checklist

- [ ] I have cleaned up my commit history and squashed fixup commits.
- [x] I have followed the [Coding Conventions](../book/src/coding-conventions.md).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: book/src/application.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Building an Application
+
```
Comment (bergzand) [inline book/src/application.md:2]: I think I'm missing a rough overview what this chapter contains. A short intro of what the goal of the chapter is and what the reader can expect.

Based on the title I was expecting a step by step guide to writing new applications, but after reading it seems like a collection of RIOT-rs specifics to writing applications. Setting some expectations in a intro paragraph could help here

**Comment by bergzand** [comment]
LGTM, please squash

File: examples/blinky/src/main.rs [hunk 2]
```diff
@@ -8,16 +8,20 @@ mod pins;
 use embassy_time::{Duration, Timer};
 use riot_rs::gpio::{Level, Output};
 
+// ANCHOR: task-example-0
```
[the remark this answers is not in the document]
Comment (kaspar030) [reply examples/blinky/src/main.rs:11]: TBH, I think they hurt a bit, and go against "minimize examples" as in, "minimize cruft".

At least in blinky and hello-world, "*the* thing everyone looks first...

[the remark this answers is not in the document]
Comment (kaspar030) [reply examples/blinky/src/main.rs:11]: So to unblock this, please copy&paste the corresponding lines for this PR, and we discuss this next weekly?

**Review by kaspar030 — APPROVED** [review]

**Pull request merged**